### PR TITLE
[SPARK-47502][INFRA] Make output the installation packages in `descending size`, add `titles`, and remove `unused` packages

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to:
- make the `size` of the `installed packages` output in the script `free_disk_space` and `free_disk_space_container` be output in `descending order`, and add `a header`.
- remove some unused installation packages in the container.

### Why are the changes needed?
Make the output content more `intuitive` and `natural`.

- Before:
  image

- After:
  image

### Does this PR introduce _any_ user-facing change?
Yes, only for dev.

### How was this patch tested?
- Pass GA.
- Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @zhengruifeng @dongjoon-hyun @HyukjinKwon

File: dev/free_disk_space_container [hunk 1]
```diff
@@ -21,13 +21,24 @@ echo "=================================="
...
+printf "Installed-Size\tPackage\n"
+dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -n -r | head -n 100
 df -h
 
 echo "Removing large packages"
 rm -rf /__t/CodeQL
 rm -rf /__t/go
 rm -rf /__t/node
 
+apt-get remove --purge -y '^aspnet.*'
```
Comment (dongjoon-hyun) [inline dev/free_disk_space_container:34]: It seems to be out of scope of the PR title, @panbingkun . Could you revise the PR title?

Comment (panbingkun) [reply dev/free_disk_space_container:34]: Done, but the title is `too long` and it's not easy to summarize it into one simple sentence, 😄

Comment (dongjoon-hyun) [reply dev/free_disk_space_container:34]: Thank you for updating.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @panbingkun .
Merged to master.

File: dev/free_disk_space_container [hunk 2]
```diff
@@ -21,13 +21,24 @@ echo "=================================="
...
+apt-get remove --purge -y '^dotnet-.*'
+apt-get remove --purge -y '^llvm-.*'
+apt-get remove --purge -y 'php.*'
+apt-get remove --purge -y '^mongodb-.*'
+apt-get remove --purge -y snapd google-chrome-stable microsoft-edge-stable firefox
+apt-get remove --purge -y azure-cli google-cloud-sdk mono-devel powershell libgl1-mesa-dri
+apt-get autoremove --purge -y
+apt-get clean
```
Comment (zhengruifeng) [inline dev/free_disk_space_container:42]: for the `apt-get remove` commands, maybe we should put them in `dev/infra/Dockerfile` ?

Comment (panbingkun) [reply dev/free_disk_space_container:42]: This is a good suggestion.  Let me give it a try.
